app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -840,27 +840,29 @@ define([
                 }
             );
 
-            if (result.oldPrice.amount !== result.finalPrice.amount) {
-                $(this.options.slyOldPriceSelector).show();
-            } else {
-                $(this.options.slyOldPriceSelector).hide();
-            }
+            if (typeof(result) != "undefined") {
+                if (result.oldPrice.amount !== result.finalPrice.amount) {
+                    $(this.options.slyOldPriceSelector).show();
+                } else {
+                    $(this.options.slyOldPriceSelector).hide();
+                }
 
-            if (result.tierPrices.length) {
-                if (this.options.tierPriceTemplate) {
-                    tierPriceHtml = mageTemplate(
-                        this.options.tierPriceTemplate,
-                        {
-                            'tierPrices': result.tierPrices,
-                            '$t': $t,
-                            'currencyFormat': this.options.jsonConfig.currencyFormat,
-                            'priceUtils': priceUtils
-                        }
-                    );
-                    $(this.options.tierPriceBlockSelector).html(tierPriceHtml).show();
+                if (result.tierPrices.length) {
+                    if (this.options.tierPriceTemplate) {
+                        tierPriceHtml = mageTemplate(
+                            this.options.tierPriceTemplate,
+                            {
+                                'tierPrices': result.tierPrices,
+                                '$t': $t,
+                                'currencyFormat': this.options.jsonConfig.currencyFormat,
+                                'priceUtils': priceUtils
+                            }
+                        );
+                        $(this.options.tierPriceBlockSelector).html(tierPriceHtml).show();
+                    }
+                } else {
+                    $(this.options.tierPriceBlockSelector).hide();
```

Comment:
If `result.tierPrices.length != 0` and `!this.options.tierPriceTemplate`, then `this.options.tierPriceBlockSelector` will be neither shown nor hidden. Is this the desired behavior?